### PR TITLE
Create advanced.md

### PR DESCRIPTION
**Pull request by zhangx501**
>^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> Remember to delete this note before submitting your pull request.
>
> For pull requests on 1.15 Features: set Milestone to 1.15 and Base Branch to dev-1.15
> 
> For pull requests on Chinese localization, set Base Branch to release-1.14
>
> For pull requests on Korean Localization: set Base Branch to dev-1.14-ko.\<latest team milestone>
>
> If you need Help on editing and submitting pull requests, visit:
> https://kubernetes.io/docs/contribute/start/#improve-existing-content.
>
> If you need Help on choosing which branch to use, visit:
> https://kubernetes.io/docs/contribute/start#choose-which-git-branch-to-use.
>^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>

### CODE REVIEW

**Comment by zhangx501** [comment]
/assign @dchen1107
Please give this PR approval.

File: content/zh/docs/user-journeys/users/app-developer/advanced.md [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+
+<!--
+#### Container-level features
+-->
+#### 容器级功能
+
+<!--
+As you may know, it's an antipattern to migrate an entire app (e.g. containerized Rails app, MySQL database, and all) into a single Pod. That being said, there are some very useful patterns that go beyond a 1:1 correspondence between a container and its Pod:
+-->
+就像你知道的那样，将整个应用程序（例如，容器化的 Rails 应用程序，MySQL 数据库和所有应用程序）迁移到单个 Pod 中是一种反模式。话虽如此，有一些非常有用的模式超出了容器和它的 pod 之间1:1的对应关系。
```
Comment (chenrui333) [inline content/zh/docs/user-journeys/users/app-developer/advanced.md]:
```suggestion
就像你知道的那样，将整个应用程序（例如，容器化的 Rails 应用程序，MySQL 数据库和所有应用程序）迁移到单个 Pod 中是一种反模式。话虽如此，有一些非常有用的模式超出了容器和它的 pod 之间 1:1 的对应关系。
```

File: content/zh/docs/user-journeys/users/app-developer/advanced.md [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+<!--
+* **Pod Presets** - Normally, to mount runtime requirements (such as environmental variables, ConfigMaps, and Secrets) into a resource, you specify them in the resource's configuration file. {{< link text="PodPresets" url="/docs/concepts/workloads/pods/podpreset/" >}} allow you to dynamically inject these requirements instead, when the resource is created. For instance, this allows team A to mount any number of new Secrets into the resources created by teams B and C, without requiring action from B and C. {{< link text="See an example" url="/docs/tasks/inject-data-application/podpreset/" >}}.
+-->
+* **Pod 预设s** -通常，将运行时的要求安装到资源中（比如环境变量、ConfigMaps、和 Secrets），请在资源的配置文件中指定它们。{{< link text="PodPresets" url="/docs/concepts/workloads/pods/podpreset/" >}}允许您在创建资源时动态地注入这些需求。例如，这允许团队 A 将任意数量的新 Secrets 安装到团队 B 和 C 创建的资源中，而无需 B 和 C 的命令操作。{{< link text="See an example" url="/docs/tasks/inject-data-application/podpreset/" >}}。
```
Comment (chenrui333) [inline content/zh/docs/user-journeys/users/app-developer/advanced.md]:
```suggestion
* **Pod 预设** -通常，将运行时的要求安装到资源中（比如环境变量、ConfigMaps、和 Secrets），请在资源的配置文件中指定它们。{{< link text="PodPresets" url="/docs/concepts/workloads/pods/podpreset/" >}}允许您在创建资源时动态地注入这些需求。例如，这允许团队 A 将任意数量的新 Secrets 安装到团队 B 和 C 创建的资源中，而无需 B 和 C 的命令操作。{{< link text="See an example" url="/docs/tasks/inject-data-application/podpreset/" >}}。
```

File: content/zh/docs/user-journeys/users/app-developer/advanced.md [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+<!--
+#### Additional API Objects
+-->
+#### 其他 API 对象
+
+<!--
+Before setting up the following resources, check to see if they are the responsibility of your organization's {{< glossary_tooltip text="cluster operators" term_id="cluster-operator" >}}.
+-->
+{{< note  >}}
+在设置以下资源之前，请检查您的组织是否负责{{< glossary_tooltip text="cluster operators" term_id="cluster-operator" >}}。
```
Comment (chenrui333) [inline content/zh/docs/user-journeys/users/app-developer/advanced.md]:
```suggestion
在设置以下资源之前，请检查您的组织是否负责{{< glossary_tooltip text="集群操作" term_id="cluster-operator" >}}。
```

File: content/zh/docs/user-journeys/users/app-developer/advanced.md [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+{{< /note  >}}
+<!--
+* **{{< glossary_tooltip text="Horizontal Pod Autoscaler (HPA)" term_id="horizontal-pod-autoscaler" >}}** - These resources are a great way to automate the process of scaling your application when CPU usage or other {{< link text="custom metrics" url="https://github.com/kubernetes/community/blob/master/contributors/design-proposals/instrumentation/custom-metrics-api.md" >}} spike. {{< link text="See an example" url="/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/" >}} to understand how HPAs are set up.
+-->
+* **{{< glossary_tooltip text="Horizontal Pod Autoscaler (HPA)" term_id="horizontal-pod-autoscaler" >}}** - 这些资源是在 CPU 使用或其他{{< link text="custom metrics" url="https://github.com/kubernetes/community/blob/master/contributors/design-proposals/instrumentation/custom-metrics-api.md" >}}峰值时自动执行缩放应用程序的好方法。
+{{< link text="See an example" url="/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/" >}}以了解如何设置 HPA。
```
Comment (chenrui333) [inline content/zh/docs/user-journeys/users/app-developer/advanced.md]:
```suggestion
{{< link text="参看一个例子" url="/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/" >}}以了解如何设置 HPA。
```

File: content/zh/docs/user-journeys/users/app-developer/advanced.md [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+<!--
+* **Federated cluster objects** - If you are running an application on multiple Kubernetes clusters using *federation*, you need to deploy the federated version of the standard Kubernetes API objects. For reference, check out the guides for setting up {{< link text="Federated ConfigMaps" url="/docs/tasks/administer-federation/configmap/" >}} and {{< link text="Federated Deployments" url="/docs/tasks/administer-federation/deployment/" >}}.
+-->
+* **联合集群对象** - 如果您在多个 Kubernetes 集群上使用*federation*运行应用程序，您需要部署标准 Kubernetes API 对象的联合版本。有关参考，请查看设置指南{{< link text="Federated ConfigMaps" url="/docs/tasks/administer-federation/configmap/" >}} and {{< link text="Federated Deployments" url="/docs/tasks/administer-federation/deployment/" >}}。
```
Comment (chenrui333) [inline content/zh/docs/user-journeys/users/app-developer/advanced.md]:
```suggestion
* **联邦集群对象** - 如果您在多个 Kubernetes 集群上使用*federation*运行应用程序，您需要部署标准 Kubernetes API 对象的联合版本。有关参考，请查看设置指南{{< link text="联邦 ConfigMaps" url="/docs/tasks/administer-federation/configmap/" >}} and {{< link text="联邦 Deployments" url="/docs/tasks/administer-federation/deployment/" >}}。
```

File: content/zh/docs/user-journeys/users/app-developer/advanced.md [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+  {{< /note  >}}
+<!--  
+#### Create Custom Resources
+-->
+#### 创建自定义资源
+
+<!-- 
+Based on the ideas above, you can define a new {{< link text="Custom Resource" url="/docs/concepts/api-extension/custom-resources/#custom-resources" >}} that is just as legitimate as a Deployment. For example, you might want to define a `Backup` object for periodic backups, if `CronJobs` don't provide all the functionality you need.
+-->
+基于以上想法，您可以定义一个新的{{< link text="Custom Resource" url="/docs/concepts/api-extension/custom-resources/#custom-resources" >}}作为合理的部署。例如，如果`CronJobs`没有提供您需要的所有功能，您可能希望为定期备份定义一个`Backup`对象。
```
Comment (chenrui333) [inline content/zh/docs/user-journeys/users/app-developer/advanced.md]:
```suggestion
基于以上想法，您可以定义一个新的{{< link text="自定义资源" url="/docs/concepts/api-extension/custom-resources/#custom-resources" >}}作为合理的部署。例如，如果`CronJobs`没有提供您需要的所有功能，您可能希望为定期备份定义一个`Backup`对象。
```

File: content/zh/docs/user-journeys/users/app-developer/advanced.md [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+
+<!-- 
+There are two main ways of setting up custom resources:
+1. **Custom Resource Definitions (CRDs)** - This method requires the least amount of implementation work. See {{< link text="an example" url="/docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions/" >}}.
+2. **API aggregation** - This method requires some {{< link text="pre-configuration" url="/docs/tasks/access-kubernetes-api/configure-aggregation-layer/" >}} before you actually {{< link text="set up a separate, extension API server" url="/docs/tasks/access-kubernetes-api/setup-extension-api-server/" >}}.
+-->
+设置自定义资源有两种主要方式：
+1.**自定义资源定义 (CRDs)** - 此方法需要的实施工作量最少。请参阅{{< link text="an example" url="/docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions/" >}}。
+2.**API 聚合** - 此方法需要一些{{< link text="pre-configuration" url="/docs/tasks/access-kubernetes-api/configure-aggregation-layer/" >}}在你实际上{{< link text="set up a separate, extension API server" url="/docs/tasks/access-kubernetes-api/setup-extension-api-server/" >}}之前。
```
Comment (chenrui333) [inline content/zh/docs/user-journeys/users/app-developer/advanced.md]:
```suggestion
2.**API 聚合** - 此方法需要一些{{< link text="pre-configuration" url="/docs/tasks/access-kubernetes-api/configure-aggregation-layer/" >}}在你实际上{{< link text="设置一个单独的 API 服务器" url="/docs/tasks/access-kubernetes-api/setup-extension-api-server/" >}}之前。
```

**Comment by zhangx501** [comment]
@chenrui333  我已经把错误的地方都改正过来了。

**Comment by zhangx501** [comment]
/assign @dchen1107 
Please give this PR approval.

**Comment by markthink** [comment]
@zhangx501 建议本地构建一下，解决构建问题

**Pull request closed without merging**